public/app/features/plugins/extensions/registry/types.ts

```diff
@@ -0,0 +1,15 @@
+import { AddedComponentRegistryItem, AddedComponentsRegistry } from './AddedComponentsRegistry';
+import { AddedLinkRegistryItem, AddedLinksRegistry } from './AddedLinksRegistry';
+import { ExposedComponentsRegistry } from './ExposedComponentsRegistry';
+import { RegistryType } from './Registry';
+
+export type PluginRegistryStates = {
+  addedComponentsRegistry: RegistryType<Array<AddedComponentRegistryItem<{}>>>;
```

Comment:
**Nit:** as the generic for `AddedComponentRegistryItem` defaults to `{}`, could we just write `RegistryType<AddedComponentRegistryItem[]>`?